From bd97485e5452e5caf8cbc439c36b7c59d89fbc66 Mon Sep 17 00:00:00 2001
From: Eric Bentley <eric.bentley@lairdtech.com>
Date: Wed, 17 Feb 2016 22:53:55 -0500
Subject: [PATCH] laird: ath6kl: remove MATCH_SSID_FLAG

the 6004 firmware does not utilize this flag.  Using it in the driver
will cause the radio to fail to send directed probes.  (this is only
seen when configured for WPA1)

Removing from the driver did not cause a noticeable deficit with the
6003 firmware, and allows the 6004 firmware to send directed probes.

bug 8942
---
 drivers/net/wireless/ath/ath6kl/cfg80211.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/drivers/net/wireless/ath/ath6kl/cfg80211.c b/drivers/net/wireless/ath/ath6kl/cfg80211.c
index 6a19cdae201c..0e23f03d959b 100644
--- a/drivers/net/wireless/ath/ath6kl/cfg80211.c
+++ b/drivers/net/wireless/ath/ath6kl/cfg80211.c
@@ -943,8 +943,6 @@ static int ath6kl_set_probed_ssids(struct ath6kl *ar,
 		else
 			ssid_list[i].flag = ANY_SSID_FLAG;
 
-		if (n_match_ssid == 0)
-			ssid_list[i].flag |= MATCH_SSID_FLAG;
 	}
 
 	index_to_add = i;
@@ -958,7 +956,6 @@ static int ath6kl_set_probed_ssids(struct ath6kl *ar,
 			    (!memcmp(ssid_list[j].ssid.ssid,
 				     match_set[i].ssid.ssid,
 				     match_set[i].ssid.ssid_len))) {
-				ssid_list[j].flag |= MATCH_SSID_FLAG;
 				ssid_found = true;
 				break;
 			}
@@ -975,7 +972,6 @@ static int ath6kl_set_probed_ssids(struct ath6kl *ar,
 		memcpy(ssid_list[index_to_add].ssid.ssid,
 		       match_set[i].ssid.ssid,
 		       match_set[i].ssid.ssid_len);
-		ssid_list[index_to_add].flag |= MATCH_SSID_FLAG;
 		index_to_add++;
 	}
 
-- 
2.11.0

